cmake/KokkosConfigCommon.cmake.in

```diff
@@ -6,6 +6,17 @@ SET(Kokkos_CXX_COMPILER "@CMAKE_CXX_COMPILER@")
 SET(Kokkos_CXX_COMPILER_ID "@KOKKOS_CXX_COMPILER_ID@")
 SET(Kokkos_CXX_STANDARD @KOKKOS_CXX_STANDARD@)
 
+# Required to be a TriBITS-compliant external package
+IF(NOT TARGET Kokkos::all_libs)
+  # CMake Error at <prefix>/lib/cmake/Kokkos/KokkosConfigCommon.cmake:10 (ADD_LIBRARY):
+  #   ADD_LIBRARY cannot create ALIAS target "Kokkos::all_libs" because target
+  #   "Kokkos::kokkos" is imported but not globally visible.
+  IF(CMAKE_VERSION VERSION_LESS "3.18")
```

Comment:
How did I determine the cmake version?
When I submitted the original version, I had tested with my local setup that has v3.25.
I was able to reproduce the CI failures with cmake-3.16 (in the CUDA RDC build that does Kokkos_INSTALL_TESTING).
After reading https://discourse.cmake.org/t/alias-and-not-globally-visible-targets/6305 I tried 3.23.5 and it did not yield an error.  Then I arbitrarily picked 3.18.0 which also worked followed by 3.17.0 that did not.

I expect this entry from the [cmake 3.18 changelog](https://cmake.org/cmake/help/latest/release/3.18.html#commands) is what resolved the issue.

> * The [add_library()](https://cmake.org/cmake/help/latest/command/add_library.html#command:add_library) and [add_executable()](https://cmake.org/cmake/help/latest/command/add_executable.html#command:add_executable) commands learned to create [Alias Targets](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#alias-targets) referencing non-GLOBAL [Imported Targets](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#imported-targets).